static/src/javascripts/projects/common/modules/identity/global/sign-in-engagement-banner.js

```diff
@@ -194,22 +202,19 @@ const show = (): void => {
 };
 
 const signInEngagementBannerInit = (): Promise<void> =>
-    canShow()
-        .then((shouldDisplay: boolean) => {
-            if (!shouldDisplay) {
-                throw new Error(ERR_EXPECTED_NO_BANNER);
-            }
-        })
-        .then(() => {
-            show();
-        })
-        .catch(err => {
-            if (err.message !== ERR_EXPECTED_NO_BANNER) throw err;
-        });
+    canShow().then((shouldDisplay: boolean) => {
+        if (shouldDisplay) show();
+    });
 
 /* this needs to be a side effect */
 recordSessionVisit();
 
+export default {
+    id: 'signInEngagementBanner',
+    show,
+    canShow,
+};
```

Comment:
Is this default export used anywhere?

Reply:
It will be from #19591. Wanted to avoid a nightmarish merge!

Reply:
Cool, makes sense. 

We lint against using `export default`, so in the long term it would be best if these moved into the named exports. Would it be possible to add a TODO comment reminding us to do this when the banner picker is set live?

Reply:
That might be an issue for the banner picker itself then as at this moment it uses the default export.

perhaps we could export a named banner object? We can import the banner type from the picker itself to enforce the contents too @GHaberis

Reply:
Thanks @walaura, makes sense, let's leave it alone